fuel-asm/tests/encoding.rs

```diff
@@ -11,261 +10,188 @@ fn opcode() {
 
-    let data_p = Opcode::from_bytes_iter(bytes.clone());
-    let data_q = Instruction::from_bytes_iter(bytes.clone());
-    let data_q: Vec<Opcode> = data_q.into_iter().collect();
+    let instructions_from_bytes: Result<Vec<Instruction>, _> = fuel_asm::from_bytes(bytes.iter().copied()).collect();
 
-    assert_eq!(data, data_p);
-    assert_eq!(data, data_q);
+    assert_eq!(instructions, instructions_from_bytes.unwrap());
 
     let pairs = bytes.chunks(8).map(|chunk| {
         let mut arr = [0; core::mem::size_of::<Word>()];
         arr.copy_from_slice(chunk);
-        Instruction::parse_word(Word::from_be_bytes(arr))
+        Word::from_be_bytes(arr)
     });
 
-    let result: Vec<Opcode> = pairs
+    let instructions_from_words: Vec<Instruction> = pairs
         .into_iter()
-        .flat_map(|(a, b)| [Opcode::from(a), Opcode::from(b)])
+        .flat_map(raw_instructions_from_word)
+        .map(|raw| Instruction::try_from(raw).unwrap())
         .collect();
 
-    assert_eq!(data, result);
-
-    let mut bytes: Vec<u8> = vec![];
-    let mut buffer = [0u8; Opcode::LEN];
-
-    for mut op in data.clone() {
-        let _ = op.read(&mut buffer).expect("Failed to write opcode to buffer");
-        bytes.extend(buffer);
-
-        let op_p = u32::from(op);
-        let op_bytes = op_p.to_be_bytes().to_vec();
-
-        let ins = Instruction::from(op_p);
-        let ins_p = Instruction::from(op);
-
-        assert_eq!(ins, ins_p);
-
-        let (_, _, _, _, _, imm_ins) = ins.into_inner();
-        let imm_op = op.immediate().unwrap_or_default();
-
-        assert_eq!(imm_op, imm_ins);
-
-        let op_p = Opcode::from(op_p);
-        let op_q = unsafe { Opcode::from_bytes_unchecked(op_bytes.as_slice()) };
-
-        assert_eq!(op, Opcode::from(ins));
-        assert_eq!(op, op_p);
-        assert_eq!(op, op_q);
-
-        let mut op_bytes = op.to_bytes().to_vec();
-
-        // Assert opcode can be created from big slices
-        op_bytes.extend_from_slice(&[0xff; 25]);
-        while op_bytes.len() > Opcode::LEN {
-            op_bytes.pop();
-
-            let op_r = unsafe { Opcode::from_bytes_unchecked(op_bytes.as_slice()) };
-            let op_s = Opcode::from_bytes(op_bytes.as_slice()).expect("Failed to safely generate op from bytes!");
-
-            assert_eq!(op, op_r);
-            assert_eq!(op, op_s);
-
-            let ins_r = unsafe { Instruction::from_slice_unchecked(op_bytes.as_slice()) };
-            let ins_s = Instruction::from_bytes(op_bytes.as_slice()).expect("Failed to safely generate op from bytes!");
-
-            assert_eq!(op, Opcode::from(ins_r));
-            assert_eq!(op, Opcode::from(ins_s));
-        }
-
-        // Assert no panic with checked function
-        while !op_bytes.is_empty() {
-            op_bytes.pop();
-
-            assert!(Opcode::from_bytes(op_bytes.as_slice()).is_err());
-        }
-
-        #[cfg(feature = "serde")]
```

Comment:
Is this removed serde coverage tested elsewhere now?

Reply:
@mitchmindtree ^

Reply:
Ahh no it's not! I think I got rid of the rest of this loop as it's not really relevant now we've the added type safety, but must have accidentally overlooked this bincode sanity check. I've pushed a commit that re-adds it.